TST: allow superseded tests failing with PROJ latest

Related: #1261
https://github.com/pyproj4/pyproj/actions/runs/4662975345/jobs/8253890884
```
FAILED test/test_transformer.py::test_transformer_group_allow_superseded_filter[6319-5703-0-2] - assert 0 == 2
 +  where 0 = len([])
 +    where [] = <TransformerGroup: best_available=False>\n- transformers: 0\n- unavailable_operations: 2.transformers
FAILED test/test_transformer.py::test_transformer_group_allow_superseded_filter[6319-5703-False-2] - assert 0 == 2
 +  where 0 = len([])
 +    where [] = <TransformerGroup: best_available=False>\n- transformers: 0\n- unavailable_operations: 2.transformers
FAILED test/test_transformer.py::test_transformer_group_allow_superseded_filter[6319-5703-True-3] - assert 0 == 3
 +  where 0 = len([])
 +    where [] = <TransformerGroup: best_available=False>\n- transformers: 0\n- unavailable_operations: 3.transformers
FAILED test/test_transformer.py::test_transformer_group_allow_superseded_filter[6319-5703-123456789-3] - assert 0 == 3
 +  where 0 = len([])
 +    where [] = <TransformerGroup: best_available=False>\n- transformers: 0\n- unavailable_operations: 3.transformers
FAILED test/test_transformer.py::test_transformer_group_allow_superseded_filter[6319-5703-blah-3] - assert 0 == 3
 +  where 0 = len([])
 +    where [] = <TransformerGroup: best_available=False>\n- transformers: 0\n- unavailable_operations: 3.transformers
============ 5 failed, 992 passed, 1 skipped, 18 warnings in 41.14s ============
```

--- a/test/test_transformer.py
+++ b/test/test_transformer.py
@@ -1654,11 +1654,8 @@ def test_transformer_group_allow_ballpark_filter():
 @pytest.mark.parametrize(
     "from_crs, to_crs, allow_superseded, expected_num_transformers",
     [
-        (6319, 5703, 0, 2),
-        (6319, 5703, False, 2),
-        (6319, 5703, True, 3),
-        (6319, 5703, 123456789, 3),
-        (6319, 5703, "blah", 3),
+        (4203, 4326, False, 4),
+        (4203, 4326, True, 5),
     ],
 )
 def test_transformer_group_allow_superseded_filter(

